packages/fxa-payments-server/src/lib/sentry.js

```diff
@@ -0,0 +1,162 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import _ from 'underscore';
+import Logger from './logger';
+import * as Sentry from '@sentry/browser';
+import Url from './url';
+
+var ALLOWED_QUERY_PARAMETERS = [
+  'automatedBrowser',
+  'client_id',
+  'context',
+  'entrypoint',
+  'keys',
+  'migration',
+  'redirect_uri',
+  'scope',
+  'service',
+  'setting',
+  'style',
+];
+
+/**
+ * function that gets called before data gets sent to error metrics
+ *
+ * @param {Object} data
+ *  Error object data
+ * @returns {Object} data
+ *  Modified error object data
+ * @private
+ */
+function beforeSend(data) {
+  if (data && data.request) {
+    if (data.request.url) {
+      data.request.url = cleanUpQueryParam(data.request.url);
+    }
+
+    if (data.tags) {
+      // if this is a known errno, then use grouping with fingerprints
+      // Docs: https://docs.sentry.io/hosted/learn/rollups/#fallback-grouping
+      if (data.tags.errno) {
+        data.fingerprint = ['errno' + data.tags.errno];
+        // if it is a known error change the error level to info.
+        data.level = 'info';
+      }
+    }
+
+    if (data.exception && data.exception.values) {
+      _.each(data.exception.values, function(value) {
+        if (value.stacktrace && value.stacktrace.frames) {
+          _.each(value.stacktrace.frames, function(frame) {
+            if (frame.abs_path) {
+              // clean up query parameters in absolute paths
+              frame.abs_path = cleanUpQueryParam(frame.abs_path); //eslint-disable-line camelcase
+            }
+          });
+        }
+      });
+    }
+
+    if (data.request.headers && data.request.headers.Referer) {
+      data.request.headers.Referer = cleanUpQueryParam(
+        data.request.headers.Referer
+      );
+    }
+  }
+
+  return data;
+}
+
+/**
+ * Overwrites sensitive query parameters with a dummy value.
+ *
+ * @param {String} url
+ * @returns {String} url
+ * @private
+ */
+function cleanUpQueryParam(url = '') {
+  var startOfParams = url.indexOf('?');
+  var newUrl = url;
+  var params;
+
+  if (startOfParams === -1) {
+    return newUrl;
+  }
+
+  params = Url.searchParams(url.substring(startOfParams + 1));
```

Comment:
It looks like here and a few lines below are the only spots where we use the new `Url` module - and that module is the only use of the new `underscore` dependency.

Could we instead use [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) from web APIs here? I think [there's a working polyfill](https://github.com/jerrybendy/url-search-params-polyfill) if we absolutely have to support browsers [older than Firefox 29](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#Browser_compatibility)

Alternatively, maybe tweak the Url module to *not* require underscore? Mainly it would be nice not to add it if we're not using it elsewhere on payments-server

Reply:
I agree, I'd like to avoid adding dependencies also. However, I filed https://github.com/mozilla/fxa/issues/3563 to consolidate these utils into fxa-shared. What do you think about addressing these changes in one place when we deal with that issue?

Reply:
Actually, now that the tests pass we still aren't meeting the coverage threshold because of the new code, maybe it does make sense to take care of this now instead of lowering the threshold. 
```
Jest: "global" coverage threshold for statements (88%) not met: 83.51%
Jest: "global" coverage threshold for branches (94%) not met: 80.99%
Jest: "global" coverage threshold for lines (88%) not met: 83.46%
```

Reply:
> maybe it does make sense to take care of this now instead of lowering the threshold.

Yeah, the thresholds are there to force us to write tests to keep the coverage quality up. Better not to cheat there 😞